indexing-service/src/main/java/io/druid/indexing/common/Counters.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.indexing.common;
+
+import com.google.common.util.concurrent.AtomicDouble;
+
+import javax.annotation.Nullable;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.BinaryOperator;
+
+public class Counters
+{
+  private final ConcurrentMap<String, AtomicInteger> intCounters = new ConcurrentHashMap<>();
+  private final ConcurrentMap<String, AtomicDouble> doubleCounters = new ConcurrentHashMap<>();
+  private final ConcurrentMap<String, AtomicReference> objectCounters = new ConcurrentHashMap<>();
```

Comment:
it seems only intCounters is used.

Reply:
I designed this to be used for any purposes. At least in the next PR for parallel indexing supporting perfect rollup, the supervisor task would collect HLL using `objectCounters`. `doubleCounters` or `intCounters` can be used for collecting ingestion metrics as well.

Reply:
ok, comment came because it looks like dead code at this point.